GDJS/Runtime/runtimegame.ts

```diff
@@ -368,19 +390,20 @@ namespace gdjs {
      * @param sceneName The name of the scene. If not defined, the first scene will be returned.
      * @return The data associated to the scene.
      */
-    getSceneData(sceneName?: string): LayoutData | null {
-      let scene: LayoutData | null = null;
+    getSceneData(sceneName?: string): LayoutAndExtensionsData | null {
       for (let i = 0, len = this._data.layouts.length; i < len; ++i) {
-        const sceneData = this._data.layouts[i];
+        const sceneData = this._data.layouts[i] as LayoutAndExtensionsData;
         if (sceneName === undefined || sceneData.name === sceneName) {
-          scene = sceneData;
-          break;
+          if (!sceneData.usedExtensionsWithVariablesData) {
+            sceneData.usedExtensionsWithVariablesData = this._data.eventsFunctionsExtensions.filter(
+              (extensionData) => extensionData.sceneVariables.length > 0
+            );
+          }
+          return sceneData;
```

Comment:
See my other comment: I would be more ok with a `{ sceneData, usedExtensionsWithVariablesData }`